tests/halos_test.py

```diff
@@ -368,10 +368,15 @@ def test_long_iord_to_pos_map():
     iord = np.array([0, 20, 10, 300])
     iord_to_fpos = halo.details.iord_mapping.make_iord_to_offset_mapper(iord)
     assert isinstance(iord_to_fpos, halo.details.iord_mapping.IordToOffsetSparse)
-
     assert (iord_to_fpos.map_ignoring_order([0, 10, 20, 300]) == np.array([0, 2, 1, 3])).all()
     assert iord_to_fpos.map_ignoring_order(300) == 3
 
+def test_long_iord_to_pos_map_non_unique():
+    with pytest.warns(UserWarning): # Warning will be raised because iord are not unique anymore
+        iord = np.array([0, 20, 20, 300])
```

Comment:
test should be constructed to be more specific about which bit of this should raise the warning. This reflects also my point above about when/where duplicate iords might arise.